t/fetcher.t

```diff
@@ -1,20 +1,273 @@
+        is scalar keys %{$data->{$source_name}{'names_list'}}, 32, "Number of names matches the content of the sample file";
+
+        my $dataset = $data->{$source_name}->{names_list};
+
+        my $aka_names = [
+            'HAFIZ SAHIB',
+            'Hafez Mohammad SAYEED',
+            'Hafiz Muhammad SAEED',
+            'Muhammad SAEED HAFIZ',
+            'Muhammad SAEED',
+            'Hafiz Mohammad SAYED',
+            'Hafiz Mohammad SAYID',
+            'Hafiz Mohammad SAEED',
+            'Hafiz Mohammad SYEED',
+            'TATA JI',
+        ];
+
+        for my $name (@$aka_names) {
+            is_deeply $dataset->{$name},
+                {
+                'dob_epoch' => [-617760000],
+                },
+                "Alias names share the same dob information ($name)";
+        }
 
-my $data = Data::Validate::Sanctions::Fetcher::run();
-cmp_ok($data->{'HMT-Sanctions'}{updated}, '>=', 1541376000, "Fetcher::run HMT-Sanctions sanctions.yml");
+        $aka_names = [
+            'Hafiz Saeed KHAN',
+            'Hafiz Said KHAN',
+            'Shaykh Hafidh Sa\'id KHAN',
+            'Hafiz Sa\'id KHAN',
+            'Hafiz Said Muhammad KHAN',
+            'Hafiz SA\'ID',
+            'Wali Hafiz Sayid KHAN',
+            'Said Khan HAFIZ',
+            'Hafez Sayed KHAN',
+            'Sayed AHMAD',
+        ];
 
-cmp_ok($data->{'OFAC-SDN'}{updated}, '>=', 1541376000, "Fetcher::run OFAC-SDN sanctions.yml");
+        for my $name (@$aka_names) {
+            is_deeply $dataset->{$name},
+                {
+                'dob_year' => [1976, 1977, 1978, 1979],
+                },
+                "Alias names share the same dob information ($name)";
+        }
 
-cmp_ok($data->{'OFAC-Consolidated'}{updated}, '>=', 1541376000, "Fetcher::run OFAC-Consolidated sanctions.yml");
+        is_deeply $dataset->{'Hafiz SAEED'},
+            {
+            'dob_epoch' => [-617760000],
+            'dob_year'  => [1976, 1977, 1978, 1979],
+            },
+            'Hafiz Saeed is shared between two groups';
 
-cmp_ok(scalar keys %{$data->{'HMT-Sanctions'}{'names_list'}}, '>', 1000, "HMT-Sanctions namelist");
+        $aka_names = [
+            'Mohammad Reza NAQDI',
+            'Gholam-reza NAQDI',
+            'Mohammad Reza SHAMS',
+            'Mohammedreza NAGHDI',
+            'Mohammad Reza NAGHDI',
+            'Mohammad-Reza NAQDI',
+            'Muhammad NAQDI',
+            'Gholamreza NAQDI',
+        ];
+        for my $name (@$aka_names) {
+            is_deeply $dataset->{$name},
+                {
+                'dob_year' => [1951, 1952, 1953, 1960, 1961, 1962],
+                },
+                "Alias names share the same dob information ($name)";
+        }
 
-cmp_ok(scalar @{$data->{'HMT-Sanctions'}{'names_list'}{'ADAM Nureldine'}{'dob_epoch'}}, '>=', 1, "check ADAM Nureldine");
+        is_deeply $dataset->{'Donald Trump'},
+            {
+            'dob_text' => ['circal-1951'],
```

Comment:
Is this a typo?

Reply:
Good point! `circal` is a typo that should be corrected in the sample files. But there's a further more serious problem with it: the original value `circal/1951` has converted into `circal-1951`.

Reply:
fixed